test/validation/api/ipsec/ipsec.c

```diff
@@ -16,11 +16,21 @@
 #include "test_vectors.h"
 #include "reass_test_vectors.h"
 
+#define EVENT_CACHE_SIZE 3
```

Comment:
The commit message and the subject line could be impoved since "event caching" does not tell much. How about something like this:

```
validation: ipsec: get event only from the specified queue

When expecting an event from a certain queue, get an event only from that
queue even when the queue is a scheduled queue. Add a queue parameter to
sched_queue_deq() and return an  event only when it came from the specified
queue. If the scheduler returns an event from another queue, buffer it
locally for possible later dequeue from that queue.
```

I also think buffering would be more accurate a term than caching. Perhaps you would like to rename the relevant identifiers in the code?

Reply:
Ack. Will rename as suggested.